packages/skeleton-react/src/lib/components/Segment/Segment.tsx

```diff
@@ -1,113 +1,102 @@
 'use client';
 
-import React, { createContext, useContext, useEffect, useState } from 'react';
+import React, { createContext, useContext, useId } from 'react';
+import * as radio from '@zag-js/radio-group';
+import { useMachine, normalizeProps } from '@zag-js/react';
 import type { SegmentContextState, SegmentProps, SegmentItemsProps } from './types.js';
+import { noop } from '$lib/internal/noop.js';
 
 // Contexts ---
 
 export const SegmentContext = createContext<SegmentContextState>({
-	value: '',
-	name: '',
-	onSelectionHandler: () => {}
+	api: {} as ReturnType<typeof radio.connect>
 });
 
 // Components ---
 
 const SegmentRoot: React.FC<SegmentProps> = ({
-	value = '',
-	name = '',
 	// Root
 	base = 'inline-flex items-stretch overflow-hidden',
 	background = 'preset-outlined-surface-200-800',
 	border = 'p-2',
-	flexDirection = 'flex-row', // vertical: flex-col
 	gap = 'gap-2',
 	padding = '',
 	rounded = 'rounded-container',
 	width = '',
 	classes = '',
+	// Indicator
+	indicatorBase = 'top-[var(--top)] left-[var(--left)] w-[var(--width)] h-[var(--height)]',
+	indicatorBg = 'preset-filled',
+	indicatorRounded = 'rounded',
+	indicatorClasses = '',
 	// Events
-	onChange,
+	onValueChange = noop,
 	// Children
-	children
+	children,
+	// Zag
+	...zagProps
 }) => {
-	function onSelectionHandler(newValue: string) {
-		value = newValue;
-		if (onChange) onChange(newValue);
-	}
+	const [state, send] = useMachine(
+		radio.machine({
+			id: useId(),
+			onValueChange: (details) => {
+				onValueChange(details.value);
+			}
+		}),
+		{ context: zagProps }
+	);
+	const api = radio.connect(state, send, normalizeProps);
 
 	// Set Context
-	const ctx = {
-		value,
-		name,
-		onSelectionHandler
-	};
+	const ctx = { api };
+
+	// Reactive
+	// FIXME: always returns vertical
```

Comment:
The `orientation` prop does work, but `vertical` is the default value. So if you want a `horizontal` control you need to pass that value. `segmented controls` are a type of radio group and on the web the radio groups are by default `vertical`. So this is work as expected.

Reply:
Oh boy, that's confusing since the default state of this component is horizontal. Would that mean we need to create our own custom prop and pass that to the machine?

Reply:
We really need `horizontal` to be the default, so I'm going to opt to a custom `orientation` prop the sets this as the default.